ssv/committee_runner.go

```diff
@@ -0,0 +1,482 @@
+			if !cr.BaseRunner.State.PostConsensusContainer.HasQuorum(validator, root) {
+				continue
+			}
+			// Skip if already submitted
+			if cr.HasSubmitted(role, validator) {
+				continue
+			}
+
+			// Reconstruct signature
+			share := cr.BaseRunner.Share[validator]
+			pubKey := share.ValidatorPubKey
+			sig, err := cr.BaseRunner.State.ReconstructBeaconSig(cr.BaseRunner.State.PostConsensusContainer, root,
+				pubKey[:], validator)
+			// If the reconstructed signature verification failed, fall back to verifying each partial signature
+			if err != nil {
+				for root := range rootSet {
+					cr.BaseRunner.FallBackAndVerifyEachSignature(cr.BaseRunner.State.PostConsensusContainer, root,
+						share.Committee, validator)
+				}
+				// Record the error and continue to next validators
+				anyErr = errors.Wrap(err, "got post-consensus quorum but it has invalid signatures")
+				continue
+			}
+			specSig := phase0.BLSSignature{}
+			copy(specSig[:], sig)
+
+			// Get the beacon object related to root
+			if _, exists := beaconObjects[validator]; !exists {
+				anyErr = errors.Wrap(err, "could not find beacon object for validator")
+				continue
+			}
+			if _, exists := beaconObjects[validator][root]; !exists {
+				anyErr = errors.Wrap(err, "could not find beacon object for validator")
+				continue
+			}
+			sszObject := beaconObjects[validator][root]
+
+			// Store objects for multiple submission
+			if role == types.BNRoleSyncCommittee {
+				syncMsg := sszObject.(*altair.SyncCommitteeMessage)
+				// Insert signature
+				syncMsg.Signature = specSig
+
+				syncCommitteeMessagesToSubmit[validator] = syncMsg
+
+			} else if role == types.BNRoleAttester {
+				att := sszObject.(*phase0.Attestation)
+				// Insert signature
+				att.Signature = specSig
+
+				attestationsToSubmit[validator] = att
+			}
+		}
+	}
+
+	// Submit multiple attestations
+	attestations := make([]*phase0.Attestation, 0)
+	for _, att := range attestationsToSubmit {
+		attestations = append(attestations, att)
+	}
+	if err := cr.beacon.SubmitAttestations(attestations); err != nil {
+		return errors.Wrap(err, "could not submit to Beacon chain reconstructed attestation")
+	}
+	// Record successful submissions
+	for validator := range attestationsToSubmit {
+		cr.RecordSubmission(types.BNRoleAttester, validator)
+	}
+
+	// Submit multiple sync committee
+	syncCommitteeMessages := make([]*altair.SyncCommitteeMessage, 0)
+	for _, syncMsg := range syncCommitteeMessagesToSubmit {
+		syncCommitteeMessages = append(syncCommitteeMessages, syncMsg)
+	}
+	if err := cr.beacon.SubmitSyncMessages(syncCommitteeMessages); err != nil {
+		return errors.Wrap(err, "could not submit to Beacon chain reconstructed signed sync committee")
+	}
+	// Record successful submissions
+	for validator := range syncCommitteeMessagesToSubmit {
+		cr.RecordSubmission(types.BNRoleSyncCommittee, validator)
+	}
+
+	if anyErr != nil {
+		return anyErr
```

Comment:
this means we return err even if some or all of the attestations/sc were submitted. maybe its best to compare how many were sent and return error only if none was submitted?

Reply:
We believe that, even though we could submit some attestations, if anything strange happened (as a signature that was incorrect for some duty) we should raise it as an error to be clear about processing problems.

Reply:
don't we want to be more accurate so maybe count how much submitted successfully and how much errerd?